src/Usos/Provider.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace SocialiteProviders\Usos;
+
+use SocialiteProviders\Manager\OAuth1\AbstractProvider;
+use SocialiteProviders\Manager\OAuth1\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'USOS';
```

Comment:
Does this need an oauth 2 provider?

Reply:
I'm not quite understand the question.

USOS service only provides an OAuth1.0a authentication. Of course, [there is a 'oauth2' branch](https://apps.usos.edu.pl/developers/api/services/oauth2/), but it was named as '2' only because they want to add another (second) version of their implementation of OAuth1.0, not the OAuth2.0.

If you mean "Why do you add this lines?": I've added them, because another OAuth1.0 drivers (Such as Twitter) also has this const field. If I am wrong, please leave me a suggestion, how can I improve this.

Reply:
I mean, is this class ever used? do we need it if the provider only supports v1?

Reply:
According to the [Contribution Guide page of the project](https://socialiteproviders.com/contribute/#creating-a-provider), Every auth driver must have a handler/listener and Provider class. If it is a driver for OAuth1.0 service, it also must have a Server class.

Also, i've tried to remove that class, and I've got an error: `Call to undefined method App\Helpers\UsosSocialiteExtension\Provider::mapUserToObject()`.  I am not calling this method anywhere in the driver, so this method called automatically by Laravel Socialite during the handling of callback.

Everything says, that the extending of AbstractProvider class is necessary.